contracts/lib/LicensingHelper.sol

```diff
@@ -35,11 +35,27 @@ library LicensingHelper {
     ) internal returns (uint256 licenseTermsId) {
         licenseTermsId = IPILicenseTemplate(pilTemplate).registerLicenseTerms(terms);
 
-        // Returns the license terms ID if already attached.
-        if (ILicenseRegistry(licenseRegistry).hasIpAttachedLicenseTerms(ipId, pilTemplate, licenseTermsId))
-            return licenseTermsId;
+        attachLicenseTerms(ipId, licensingModule, licenseRegistry, pilTemplate, licenseTermsId);
+    }
+
+    /// @dev Attaches license terms to the given IP.
+    /// @param ipId The ID of the IP.
+    /// @param licensingModule The address of the Licensing Module.
+    /// @param licenseRegistry The address of the License Registry.
+    /// @param licenseTemplate The address of the license template.
+    /// @param licenseTermsId The ID of the license terms to be attached.
+    function attachLicenseTerms(
+        address ipId,
+        address licensingModule,
+        address licenseRegistry,
+        address licenseTemplate,
+        uint256 licenseTermsId
+    ) internal {
+        // Returns if license terms are already attached.
+        if (ILicenseRegistry(licenseRegistry).hasIpAttachedLicenseTerms(ipId, licenseTemplate, licenseTermsId))
+            return;
 
-        ILicensingModule(licensingModule).attachLicenseTerms(ipId, pilTemplate, licenseTermsId);
+        ILicensingModule(licensingModule).attachLicenseTerms(ipId, licenseTemplate, licenseTermsId);
```

Comment:
We might consider to employ try/catch around the attachLicenseTerms() function so that don't have to call hasIpAttachedLicenserTerms() to further save the gas.

Reply:
Don't need to address in this PR, we can address gas optimization in later PR.

Reply:
- Created an issue to track this #60.  Will get back to this later :)